docs/jdbc_connector.rst

```diff
@@ -0,0 +1,376 @@
+iteration. Several modes are supported, each of which differs in how modified rows are detected.
+
+Incremental Query Modes
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Each incremental query mode tracks a set of columns for each row, which it uses to keep track of
+which rows have been processed and which rows are new or have been updated. The ``mode`` setting
+controls this behavior and supports the following options:
+
+* **Incrementing Column**: A single column containing a unique ID for each row, where newer rows are
+  guaranteed to have larger IDs, i.e. an ``AUTOINCREMENT`` column. Note that this mode can only
+  detect *new* rows. *Updates* to existing rows cannot be detected, so this mode should only be
+  used for immutable data. One example where you might use this mode is when streaming fact
+  tables in a data warehouse, since those are typically insert-only.
+
+* **Timestamp Column**: In this mode, a single column containing a modification timestamp is used
+  to track the last time data was processed and to query only for rows that have been modified
+  since that time. Note that because timestamps are no necessarily unique, this mode cannot
+  guarantee all updated data will be delivered: if 2 rows share the same timestamp and are
+  returned by an incremental query, but only one has been processed before a crash, the second
+  update will be missed when the system recovers.
+
+* **Timestamp and Incrementing Columns**: This is the most robust and accurate mode, combining an
+  incrementing column with a timestamp column. By combining the two, as long as the timestamp is
+  sufficiently granular, each (id, timestamp) tuple will uniquely identify an update to a row. Even
+  if an update fails after partially completing, unprocessed updates will are still correctly
+  detected and delivered when the system recovers.
+
+* **Custom Query**: The JDBC connector supports using custom queries instead of copying whole
+  tables. With a custom query, one of the other update automatic update modes can be used as long
+  as the necessary ``WHERE`` clause can be correctly appended to the query. Alternatively, the
+  specified query may handle filtering to new updates itself;
+  however, note that no offset tracking will be performed (unlike the automatic modes where
+  ``incrementing`` and/or ``timestamp`` column values are recorded for each record), so the query
+  must track offsets itself.
+
+* **Bulk**: This mode is unfiltered and therefore not incremental at all. It will load all rows
+  from a table on each iteration. This can be useful if you want to periodically dump an entire
+  table where entries are eventually deleted and the downstream system can safely handle duplicates.
+
+Note that all incremental query modes that use certain columns to detect changes will require
+indexes on those columns to efficiently perform the queries.
+
+For incremental query modes that use timestamps, the JDBC connector uses a configuration
```

Comment:
nit: maybe "the wait after" -> "the waiting period after". Same in following sentence